tests/RetryTest.php

```diff
@@ -202,4 +206,22 @@ public function testDeadConnectionException(): void
             $this->assertContains('Dead connection', $ue->getMessage());
         }
     }
+
+    public function testNoRetryOnAppError(): void
+    {
+        $testLogger = new Logger('common-retry-test-logger');
+        $testLogger->pushHandler(new TestHandler());
+
+        $config = $this->getRetryConfig();
+        @unlink($this->dataDir . '/out/tables/in.c-main.sales.csv');
+        touch($this->dataDir . '/out/tables/in.c-main.sales.csv');
+        chmod($this->dataDir . '/out/tables/in.c-main.sales.csv', 444);
+
+        try {
```

Comment:
i'm not sure about phpunit version here, but if self::expectException and self::expectExceptionMessage are available better use them